tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -0,0 +1,410 @@
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/structpath"
+)
+
+
+
+var (
+	vmEnv     map[string]string
+)
+
+
+
+func mustReadCert(t *testing.T, f string) string {
+	b, err := ioutil.ReadFile(path.Join(env.IstioSrc, "tests/testdata/certs/dns", f))
+	if err != nil {
+		t.Fatalf("failed to read %v: %v", f, err)
+	}
+	return string(b)
+}
+
+func TestClientToServiceTls(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.file-mounted-certs").
+		Run(func(ctx framework.TestContext) {
+
+			echoClient, echoServer, serviceNamespace := setupEcho(t, ctx)
+
+			bufDestinationRule := createObject(t, ctx, serviceNamespace.Name(), DestinationRuleConfigMutual)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufDestinationRule.String())
+
+			bufVirtualService := createObject(t, ctx,serviceNamespace.Name(), VirtualServiceConfigMutual)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufVirtualService.String())
+
+			bufPeerAuthentication := createObject(t, ctx, "istio-system", PeerAuthenticationConfig)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, "istio-system", bufPeerAuthentication.String())
+
+			bufEnvoyFilter := createObject(t, ctx, serviceNamespace.Name(), EnvoyFilterConfig)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufEnvoyFilter.String())
+
+			retry.UntilSuccessOrFail(t, func() error {
+				resp, err := echoClient.Call(echo.CallOptions{
+					Target:   echoServer,
+					PortName: "http",
+					Scheme:   scheme.HTTP,
+				})
+
+				if err != nil {
+					return fmt.Errorf("request failed: %v", err)
+				}
+
+				codes := make([]string, 0, len(resp))
+				for _, r := range resp {
+					codes = append(codes, r.Code)
+				}
+				if !reflect.DeepEqual(codes, []string{response.StatusCodeOK}) {
+					return fmt.Errorf("got codes %q, expected %q", codes, []string{response.StatusCodeOK})
+				}
+				for _, r := range resp {
+					if _, f := r.RawResponse["X-Forwarded-Client-Cert"]; !f {
+						return fmt.Errorf("expected to be see XFCC header, but none found. response: %+v", r.RawResponse)
+					}
+				}
+				return nil
+			}, retry.Delay(5*time.Second), retry.Timeout(1*time.Minute))
+
+			verifyUsingFileBasedCerts(t, echoClient, echoServer)
+			verifyStats(t, echoClient, echoServer)
+	})
+}
+
+const (
+	DestinationRuleConfigMutual = `
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: server
+  namespace: {{.AppNamespace}}
+spec:
+  host: "server.{{.AppNamespace}}.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: MUTUAL
+      caCertificates: /client-certs/ca.pem
+      clientCertificate: /client-certs/cert.pem
+      privateKey: /client-certs/key.pem
+      subjectAltNames:
+        - server.default.svc
+
+`
+	VirtualServiceConfigMutual = `
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: server
+  namespace: {{.AppNamespace}}
```

Comment:
nit: you can just apply it to AppNamespace and use short names if you want to avoid templating